pkg/agent/loadbalancer/servers.go

```diff
@@ -84,20 +91,61 @@ func (lb *LoadBalancer) nextServer(failedServer string) (string, error) {
 	return lb.currentServerAddress, nil
 }
 
-// dialContext dials a new connection, and adds its wrapped connection to the map
+// dialContext dials a new connection using the environment's proxy settings, and adds its wrapped connection to the map
 func (s *server) dialContext(ctx context.Context, network, address string) (net.Conn, error) {
-	conn, err := defaultDialer.DialContext(ctx, network, address)
+	// Check if env variable for proxy is set
```

Comment:
It seems unnecessary to check the env var and create a new dialer every time a new connection is dialed. It also adds a lot more code here to switch between the two dialers.

Why not just add an `init()` function that checks the environment variable, and if set replaces `var defaultDialer = &net.Dialer{}` with the proxy dialer?

Reply:
Makes a lot of sense, I've edited the function back to normal and exported the code to init(), let me know if anything else needs to be done.